capsules/src/virtual_i2c.rs

```diff
@@ -64,14 +94,87 @@ impl<'a> MuxI2C<'a> {
                         Op::WriteRead(wlen, rlen) => {
                             self.i2c.write_read(node.addr, buf, wlen, rlen)
                         }
+                        Op::CommandComplete(err) => {
+                            self.command_complete(buf, err);
+                        }
                         Op::Idle => {} // Can't get here...
                     }
                 });
                 node.operation.set(Op::Idle);
-                self.inflight.set(node);
+                self.i2c_inflight.set(node);
             });
+
+            if self.i2c_inflight.is_none() && self.smbus.is_some() {
```

Comment:
Don't you also need to check that `self.smbus_inflight.is_none()` here?

Reply:
We are still inside the `if self.i2c_inflight.is_none() && self.smbus_inflight.is_none() {` statement.

Reply:
Oops. Sorry about that! I didn't see the indentation level.